lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/RunningWorkers.java

```diff
@@ -143,6 +156,29 @@ void launchTasklet(final Tasklet tasklet) {
 
   /**
    * Concurrency: Called by multiple threads.
+   * Parameter: Same taskletId can come in multiple times.
+   */
+  void cancelTasklet(final boolean mayInterruptIfRunning, final int taskletId) {
+    lock.lock();
+    try {
+      final String workerId = getWhereTaskletWasScheduledTo(taskletId); // This is not ideal.
```

Comment:
Could you elaborate a bit why this is not ideal? Is it related to duplicate execution(REEF-500)?